app-hw-37/src/CardsList/Card.tsx

```diff
@@ -20,14 +21,13 @@ type Cardprops = {
 }
 
 export const Card = ({ cardinfo, showFullScreen }: Cardprops) => {
-	const handleClick = () => {
-		showFullScreen(cardinfo.id)
-	}
-	const theme = useContext(ThemeContext)
+	const dispatch = useAppDispatch()
+	const theme = useAppSelector((state) => state.theme.themeColor)
+
 	return (
-		<button className='post' id={String(cardinfo.id) + '-1'} onClick={handleClick} style={theme === 'light' ? { background: 'white' } : { background: '#423e3e' }}>
+		<button className='post' onClick={() => showFullScreen(cardinfo.id)} style={theme === 'light' ? { background: 'white' } : { background: '#423e3e' }}>
 			<div className="post__img-wrapper">
-				<img src={cardinfo.image} alt={cardinfo.text} />
+				<Link to='/posts/img'><img src={cardinfo.image} alt={cardinfo.text} onClick={() => dispatch(setCurrentImg(cardinfo.image ? cardinfo.image : '#'))} /></Link>
```

Comment:
if `cardinfo.image === undefined` prefer not to render it at all, then change onClick